esmvalcore/preprocessor/_time.py

```diff
@@ -157,22 +135,27 @@ def _duration_to_date(duration, reference, sign):
     return date
 
 
-def _restore_time_coord_position(cube, original_time_index):
-    """Restore original ordering of coordinates."""
-    # Coordinates before time
-    new_order = list(np.arange(original_time_index) + 1)
-
-    # Time coordinate
-    new_order.append(0)
-
-    # Coordinates after time
-    new_order = new_order + list(range(original_time_index + 1, cube.ndim))
-
-    # Transpose cube in-place
-    cube.transpose(new_order)
+def _select_timeslice(
+    cube: iris.cube.Cube,
+    select: np.ndarray,
+) -> iris.cube.Cube | None:
+    """Slice a cube along its time axis."""
+    if select.any():
+        coord = cube.coord('time')
+        time_dim = cube.coord_dims(coord)[0]
+        slices = tuple(select if i == time_dim else slice(None)
+                       for i in range(cube.ndim))
+        cube_slice = cube[slices]
+    else:
+        cube_slice = None
+    return cube_slice
 
 
-def _extract_datetime(cube, start_datetime, end_datetime):
+def _extract_datetime(
+    cube: iris.cube.Cube,
```

Comment:
Just wondering, is this new way of specifying input types is something that should be done from now on in new PRs?

Reply:
[They are optional](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#python), but if you find that it makes the code more clear then you are welcome to add them. Personally, started doing it if the type is not too complicated: if I need a lot of [`typing.Union`](https://docs.python.org/3/library/typing.html#typing.Union)s etc to describe the type I leave it out for now. In Python 3.10 a much nicer way of combining types was introduced `X | Y` instead of `typing.Union(X, Y)`, but in older Python versions type hints can feel quite clunky to me. A nice feature that I like about type hints, apart from the improved readability, is that my editor (vscode) shows me what methods there are available on the objects that I pass into a function, so I do not need to look this up on the API webpages anymore.